### DISCUSSION TITLE
changing background color of lualine with rose-pine

### DISCUSSION THREAD

**Discussion by Zynh0722** [rose-pine/neovim#145] [Q&A] [answered]
I'm rather new to the vim world, but I've got something so very close to being absolutely fantastic
![](https://imgur.com/a/JbQiTQ0)

Basically all I want to do is change the bg of the space behind the filename and such to `none` rather than `base`.

But honestly I've not a clue what black magic is going on for rose-pine to be an acceptable option for a lualine theme, let alone what I need to do to tinker with it.

I thinkkkk all Im trying to do is change the c section bg, which I tried to do the way lualine's README states, but that didnt work, likely because rose-pine isn't a lualine theme.

After further investigation, I tried requring `rose-pine.lualine`, boy did it not like that.

Is what I'm trying to do possible in a thus far unexplored by me avenue?

**Comment by mrtnvgr** [comment, marked as answer]
```lua
require("rose-pine").setup()

local theme = require("lualine.themes.rose-pine")
local modes = { "normal", "insert", "visual", "replace", "command", "inactive" }
for _, mode in ipairs(modes) do
  theme[mode].c.bg = "NONE"
end

require("lualine").setup({
  options = {
    theme = theme,
  },
})
```

**Reply by Zynh0722** [reply]
Well, I have tried to get that to work. 

It's almost certainly something to do with my setup, but I wasn't sure how to modify the load order of my `after/` directory

So I moved my configs for rose-pine and lualine into my packerfile, so it gets compiled into my packer-compiled

I tried using after to put lualine further into the config line, but honestly, I'm not sure why it doesn't work.

my packer lua
https://github.com/Zynh0722/nvim/blob/main/lua/zynh/packer.lua

the previous configs can be found here;
https://github.com/Zynh0722/nvim/blob/main/after/plugin/rose-pine.lua
https://github.com/Zynh0722/nvim/blob/main/after/plugin/lualine.lua

**Reply by mrtnvgr** [reply]
Actually that snippet gave me error (I wrote it on my phone without any testing):

Try this:
```lua
local custom_auto = require("lualine.themes.auto")
custom_auto.normal.c.bg = "NONE"
```

**Reply by mrtnvgr** [reply]
I looked in your lualine config. You need to set a theme in options.
```lua
            -- Set lualine as statusline
            -- See `:help lualine.txt`
            require('lualine').setup {
                options = {
                    theme = custom_auto, -- <--
                    icons_enabled = false,
                    theme = 'rose-pine',
                    component_separators = '|',
                    section_separators = '',
                },
            }
        end
    }  
```

**Comment by Zynh0722** [comment]
Oh my goodness, I totally did forget to put the customized theme in the theme spot, rather than default rose-pine.

Thank you so much! I was able to go back to my original config setup, with configs in dedicated files.

My packer file is once again clean (at least of my rose-pine and lualine configs :P)
